src/volume/inmem.rs

```diff
@@ -42,6 +42,23 @@ pub struct InMemNiftiVolume {
 }
 
 impl InMemNiftiVolume {
+    /// Build a InMemNiftiVolume from a header and a buffer. The buffer length and the dimensions
+    /// declared in the header are expected to fit.
+    pub fn from_raw_data(header: &NiftiHeader, raw_data: Vec<u8>) -> Result<Self> {
+        assert!(nb_bytes_for_data(header) == raw_data.len(),
```

Comment:
Oops, forgot about this line. Please make it a recoverable error by making a new error variant.